pmlc/dialect/stripe/autostencil_pass.cc

```diff
@@ -0,0 +1,312 @@
+// Copyright 2019, Intel Corporation
+
+#include "pmlc/dialect/stripe/transcode.h"
+#include "pmlc/dialect/stripe/autostencil_pass.h"
+
+#include <sstream>
+#include <vector>
+
+#include "mlir/IR/Matchers.h"
+#include "mlir/Translation.h"
+
+#include "pmlc/dialect/stripe/analysis.h"
+#include "pmlc/dialect/stripe/dialect.h"
+#include "pmlc/dialect/stripe/ops.h"
+#include "pmlc/dialect/stripe/transforms.h"
+#include "pmlc/dialect/stripe/util.h"
+
+#include "base/util/logging.h"
+
+namespace pmlc {
+namespace dialect {
+namespace stripe {
+
+// Number of tensors for the matrix multiplication
+const size_t numTensors = 3;
+// Number of searching index, i.e., M, N, K
+const size_t numIndex = 3;
+// Maximum number of index in a block, e.g., ParallelForOp
+const size_t indexLimit = 32;
+
+typedef llvm::SmallPtrSet<BlockArgument*, indexLimit> BlockArgumentSet;
```

Comment:
Also, let's start trying to use the new `using` syntax since it's a little easier to read:

```
using BlockArgumentSet = llvm::SmallPtrSet<BlockArgument*, indexLimit>
```